programs/util.c

```diff
@@ -54,14 +54,24 @@ int UTIL_getFileStat(const char* infilename, stat_t *statbuf)
 int UTIL_setFileStat(const char *filename, stat_t *statbuf)
 {
     int res = 0;
+#if defined(_WIN32) || (_POSIX_C_SOURCE < 200809L)
```

Comment:
So you should be checking the value of `PLATFORM_POSIX_VERSION`, not `_POSIX_C_SOURCE`, in all of these.

I also have a weak preference for making these positive checks, putting the preferred behavior first (e.g., `#if PLATFORM_POSIX_VERSION >= 200809L` etc., not `#if PLATFORM_POSIX_VERSION < 200809L`).

Reply:
PLATFORM_POSIX_VERSION >= 200809L is the wrong define here. Should be less than since it's for utime, not utimensat. Could make it > 200112L.

Reply:
Actually that's wrong too...

I'll keep it as is.